src/components/productrecommendation.fetchallpages.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { login } from '../utils/AuthService';
+import ProductListItemMain from './productlistitem.main';
+
+const Config = require('Config');
+
+
+class Productrecommendationfetchallpages extends React.Component {
+  static propTypes = {
+    product: PropTypes.objectOf(PropTypes.any).isRequired,
+    length: PropTypes.number.isRequired,
+    count: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      links: [],
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    const { links } = this.state;
+    const link = links;
```

Comment:
its a bad practice to update the state directly. Thats how most recommendations were. Additionally elist wont let you do it.

Reply:
you are correct, but you have already deconstructed links. so its not the state itself. the eslint error you're encountering is actually different. i'm assuming its when you try to set the state later as setState({links: links}) so you've changed to link so that works out. but the error its complaining wants you to setState({links})

Reply:
perfect! :)